zcash_client_sqlite/src/wallet/init/migrations.rs

```diff
@@ -157,4 +216,46 @@ mod tests {
             Ok(_)
         );
     }
+
+    #[test]
+    fn migrate_between_releases_without_data() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), Network::TestNetwork).unwrap();
+
+        let seed = [0xab; 32].to_vec();
+
+        let mut prev_state = HashSet::new();
+        let mut ensure_migration_state_changed = |conn: &Connection| {
+            let new_state = conn
+                .prepare_cached("SELECT * FROM schemer_migrations")
+                .unwrap()
+                .query_map([], |row| row.get::<_, [u8; 16]>(0).map(Uuid::from_bytes))
+                .unwrap()
+                .collect::<Result<HashSet<Uuid>, _>>()
+                .unwrap();
+            assert!(prev_state != new_state);
+            prev_state = new_state;
+        };
+
+        for migrations in super::PUBLIC_MIGRATION_STATES {
+            assert_matches!(
+                init_wallet_db_internal(
+                    &mut db_data,
+                    Some(Secret::new(seed.clone())),
+                    migrations,
```

Comment:
Does this cause the migrations to be run in a particular order? Because if so, this test would still potentially miss this bug; in order to catch it, we would need to run all of the permutations of the leaf migrations.

Reply:
The migration order permutation issue does need fixing, but not in this release.